semanticdb/semanticdb3/javacp.md

```diff
@@ -0,0 +1,551 @@
+  </tr>
+</table>
+
+Notes:
+* Class members must appear in the order specified below. This requirement is
+  necessary to compute consistent method symbol disambiguators in the Scala
+  compiler where static and non-static members have separate owners.
+  * non-static members first, following the same order as they appear in the original source
+  * static members secondly, following the same order as they appear in the original source
+* A Java class maps to a single symbol with type `ClassInfoType` including all static and non-static members.
+  This departs from the Scala compiler internal representation of Java classes where non-static members
+  are grouped under a `CLASS` symbol and static members are grouped under an `OBJECT` symbol.
+* Supported properties for `CLASS` symbols are
+  * `FINAL` set for all final classes
+  * `ABSTRACT` set for all final classes
+  * `STATIC` set for static inner classes
+  * `ENUM` set for enum types
+* Supported accessibilities for `CLASS` symbols are
+  * `PRIVATE`: set for `private` inner classes
+
+**Enum types** [\[84\]][84] are represented by a single symbol with the `CLASS` kind.
+
+```java
+package a;
+
+public enum Coin {
+  PENNY, NICKEL
+}
+```
+
+<table>
+  <tr>
+    <td><b>Definition</b></td>
+    <td width="275px"><b>Symbol</b></td>
+    <td><b>Kind</b></td>
+    <td><b>Signature</b></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#</code></td>
+    <td><code>CLASS</code></td>
+    <td><code>ClassInfoType(List(), List(&lt;Enum&lt;Coin&gt;&gt;), List(&lt;PENNY&gt;, &lt;NICKEL&gt;))</code></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#PENNY.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;Coin&gt;), List())</code></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#NICKEL.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;Coin&gt;), List())</code></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#values().</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;Array&gt;, List(&lt;Coin&gt;)))</code></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#valueOf(String).</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;Coin&gt;, List()))</code></td>
+  </tr>
+</table>
+
+Notes:
+* Enum types follow the same rules as class declarations.
+* Enum type classes and enum fields must have the property `ENUM` set.
+* Enum fields have the type of their enclosing class.
+* Enum members include the implicitly declared methods [\[86\]][86] `valueOf` and `values`.
+
+**Interface declarations** [\[77\]][77] are represented by a single symbol
+like classes but with the `INTERFACE` kind. Concretely, the differences
+between interface symbols and class symbols are:
+* Interface symbols do not support properties.
```

Comment:
Let's make interface symbols always `ABSTRACT`, like we did with Scala objects and `FINAL`?